new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/TestActions.java

```diff
@@ -38,6 +44,37 @@ public static String createUniqueNamespace() {
         return Namespace.createUniqueNamespace();
     }
 
-    // etc...
+    // ------------------------ Docker image  -------------------------
+
+    public static boolean createMIIImage(
+        String baseImageName, 
+        String imageTag,
+        List<String> modelFiles,
+        List<String> modelVariableFiles,
+        List<String> modelArchiveFiles,
+        String wdtVersion,
+        String domainType) {
+  
+    	return 
+            new WebLogicImageTool()
```

Comment:
this is ok for now, but i might ask to make it statics rather than passing around objects when i see you use this in  a test

Reply:
I guess I was thinking of something different.  I would have thought createMIIImage would take a WITParams() as a parameter.   It wouldn't have a signature with all of the params listed out - instead WITParams would be built in some central location, or generated from some central set of settings for the suite, and passed in.

Reply:
yes, some reusable object with a fluent api would make a nice param to this method

Reply:
I think the problem may be the additional layer at the WebLogicImageTool level. One solution is to remove the WITParams, and use the WebLogicImageTool as the reusable object with the fluent api that is in WITParams now, as well as the updateImage method that WebLogicImageTool currently has.

Reply:
Then the TestActions will have a with() method that takes a WebLogicImageTool object. A test code of creating a MII image would look like the following.

WebLogicImageTool wit = new WebLogicImageTool()
                                                 .imageName("xxx)
                                                 .tag("yyy")
                                                 and etc.
new TestActions().with(wit).createMIIImage().

Reply:
I can give it a try and see if it looks better.

Reply:
I think it is easier to just stick with the current approach and change the parameters into a WITParams.

Reply:
no - we would not instantiate TestActions - all of the methods are statics, so you would `import static TestActions` and then say:
`with(wit).createMIIImage()`

Reply:
I think what I eventually did is similar to what you suggested - TestActions does the following.
new WebLogicImageTool().with(wit).updateImage().

Reply:
if I understand correctly, you two are close.  But, I think Mark is suggesting an overloaded "with()" method where the argument decides the operation.  In this case "with(WebLogicImageTool)" finds the ImageTool with statement, and since the TestActions is a static import and all "with()" methods are static, it looks more like with(wit).createMIIImage().  I think you are mostly on the same page here, so approving.